ansible/include_vars.yml

```diff
@@ -1,65 +1,45 @@
 ---
-- name: Step 0000 Find Include Vars files
-  hosts: localhost
-  connection: local
+- name: Step 0000 Include Vars
+  hosts:
+  - localhost
+  - all
   gather_facts: no
   tags: include_vars
   tasks:
-  - name: Stat default variables files (both yaml/yml extensions)
+  - name: Set output_dir for all hosts
+    set_fact:
+      output_dir: "{{ hostvars.localhost.output_dir }}"
+    when: hostvars.localhost.output_dir is defined
+
+  - name: Include variables files
     vars:
-      find_me:
+      __vars_file_base:
       # Global default vars related to the cloud provider
       - cloud_providers/{{ cloud_provider }}_default_vars
       # Legacy env_vars.yml (replaced by default_vars.yml)
       - configs/{{ env_type }}/env_vars
       # Default vars of the config
       - configs/{{ env_type }}/default_vars
       # Default vars of the config, specific to a cloud provider
-      - configs/{{ env_type }}/default_vars.{{ cloud_provider }}
       - configs/{{ env_type }}/default_vars_{{ cloud_provider }}
       # Lecacy secret vars file.
       - configs/{{ env_type }}/env_secret_vars
-      extensions:
+      __vars_file_extension:
       - yaml
       - yml
-    stat:
-      path: "{{ item[0] ~ '.' ~ item[1] }}"
-    loop: "{{ find_me | product(extensions) | list }}"
-    register: rstat_varfiles
-
-  - name: Stat variables files
-    vars:
-      find_me:
-      # secret file path passed as extra-var
-      - "{{ secret_file | d('/secret/file/not/passed') }}"
-    stat:
-      path: "{{ item }}"
-    loop: "{{ find_me }}"
-    register: rstat2_varfiles
-
-- name: Step 0000 Include vars
-  hosts:
-  - localhost
-  - all
-  connection: local
-  gather_facts: no
-  tags: include_vars
-  tasks:
-  - name: Set output_dir for all hosts
-    set_fact:
-      output_dir: "{{ hostvars.localhost.output_dir }}"
-    when: hostvars.localhost.output_dir is defined
-
-  - name: Include variables files
+      __vars_file: "{{ playbook_dir }}/{{ __vars_file_parts[0] }}.{{ __vars_file_parts[1] }}"
+    loop: "{{ __vars_file_base | product(__vars_file_extension) | list }}"
+    loop_control:
+      loop_var: __vars_file_parts
+      label: "{{ __vars_file }}"
+    when:
+    - __vars_file is file
     include_vars:
-      file: "{{ item.stat.path }}"
+      file: "{{ __vars_file }}"
+
+  - name: Set passthrough user data
     when:
-    - item is not skipped
-    - item.stat.exists
-    loop: "{{ hostvars.localhost.rstat_varfiles.results + hostvars.localhost.rstat2_varfiles.results }}"
-    loop_control:
-      label: >-
-        {{ (
-          item.stat.path
-          | default('skipped')
-        ) if 'stat' in item else item }}
+    - agnosticd_passthrough_user_data is defined
+    - agnosticd_passthrough_user_data | length > 0
+    agnosticd_user_info:
+      data: "{{ agnosticd_passthrough_user_data }}"
```

Comment:
are you running this on all hosts, thus several times?
VS once if it's setup_runtime.yml ?

Reply:
The reason for the move is so that `agnosticd_passthrough_user_data` can be set from default vars.